pkg/cli/cmd/app/delete/delete_test.go

```diff
@@ -124,18 +125,18 @@ func Test_Delete(t *testing.T) {
 		defer ctrl.Finish()
 
 		appManagementClient := clients.NewMockApplicationsManagementClient(ctrl)
-
-		appManagementClient.EXPECT().
-			GetApplication(gomock.Any(), "test-app").
-			Return(v20231001preview.ApplicationResource{
-				Properties: &v20231001preview.ApplicationProperties{
-					Environment: to.Ptr("/planes/radius/local/resourceGroups/default/providers/Applications.Core/environments/another-env"),
+		deleteMock := delete.NewMockInterface(ctrl)
```

Comment:
The test setup is comprehensive. Consider adding tests for error scenarios in the progress reporting system.